tcp_logger.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"io"
+	"log"
+	"net"
+	"os"
+	"time"
+)
+
+// TCPLogger is a logging service that will spool messages in a local channel
+// that are intended to be flushed out to a net.Conn given.
+// Note that messages are not guaranteed to be delivered, but attempts are made to
+// redeliver messages given connection.Write errors.
+type TCPLogger struct {
+	conn          net.Conn
+	deadlineWait  time.Duration
+	retryLimit    int
+	retries       int
+	sleepDuration time.Duration
+
+	logLines chan []byte
+}
+
+// NewTCPLogger is used to establish a new TCPLogger.
+func NewTCPLogger(conn net.Conn, retryLimit int) TCPLogger {
+	return TCPLogger{
+		conn:          conn,
+		deadlineWait:  time.Minute,
+		retryLimit:    retryLimit,
+		sleepDuration: time.Minute,
+
+		logLines: make(chan []byte, 1000), // Should this be a buffered channel?
+	}
+}
+
+// Write pushes bytes given into local chan to flush out to connection.
+func (t *TCPLogger) Write(p []byte) (n int, err error) {
+	t.logLines <- p
```

Comment:
we should probably have a:

```
select {
case t.logLines <- p:
  return len(p), nil
default:
  log.Println("Error: Could not log to TCP output because the queue is full")
  return 0, nil // doesn't look like we care about the response
}
```

Reply:
I worry blocking on the STDOUT from journald could lead to other issues, so we should never block (if possible).

Reply:
I like it, added select with default.